### PR TITLE
HWKAPM-766 Fix issue with ruby zipkin data when server span doesn't h…

### PR DESCRIPTION
**Pull request by objectiser**
…ave URL, but client span does

### CODE REVIEW

**Comment by objectiser** [comment]
@pavolloffay Could you review please?

I've tested it against zipkin polyglot example and it now displays the roda service correctly.

**Comment by objectiser** [comment]
roda-fix

File: server/processors-zipkin/src/test/java/org/hawkular/apm/server/processor/zipkin/NodeDetailsDeriverTest.java [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+        ss.setTimestamp(1000);
+        ss.setValue("ss");
+        BinaryAnnotation httpAddress = new BinaryAnnotation();
+        httpAddress.setKey(Constants.ZIPKIN_BIN_ANNOTATION_HTTP_URL);
+        httpAddress.setValue("http://localhost/hello");
+        Span serverSpan = new Span(Arrays.asList(httpAddress), Arrays.asList(sr, ss));
+        serverSpan.setId("2");
+        serverSpan.setTraceId("1");
+        serverSpan.setTimestamp(0L);
+        serverSpan.setDuration(ss.getTimestamp() - sr.getTimestamp());
```
Comment (pavolloffay) [inline server/processors-zipkin/src/test/java/org/hawkular/apm/server/processor/zipkin/NodeDetailsDeriverTest.java]: Could you please remove this(and timestamp too) if it is not objective of this test?

File: server/processors-zipkin/src/test/java/org/hawkular/apm/server/processor/zipkin/NodeDetailsDeriverTest.java [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        sr.setValue("sr");
+        sr.setTimestamp(0);
+        Annotation ss = new Annotation();
+        ss.setTimestamp(1000);
+        ss.setValue("ss");
+        Span serverSpan = new Span(Collections.emptyList(), Arrays.asList(sr, ss));
+        serverSpan.setId("2");
+        serverSpan.setTraceId("1");
+        serverSpan.setTimestamp(0L);
+        serverSpan.setDuration(ss.getTimestamp() - sr.getTimestamp());
```
Comment (pavolloffay) [inline server/processors-zipkin/src/test/java/org/hawkular/apm/server/processor/zipkin/NodeDetailsDeriverTest.java]: timestamp and duration is not necessary

File: server/processors-zipkin/src/test/java/org/hawkular/apm/server/processor/zipkin/NodeDetailsDeriverTest.java [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+        cs.setTimestamp(0);
+        cs.setValue("cs");
+        Annotation cr = new Annotation();
+        cr.setValue("cr");
+        cr.setTimestamp(1000);
+        Span clientSpan = new Span(Collections.emptyList(), Arrays.asList(cs, cr));
+        clientSpan.setId("2");
+        clientSpan.setTraceId("1");
+        clientSpan.setTimestamp(0L);
+        clientSpan.setDuration(cr.getTimestamp() - cs.getTimestamp());
```
Comment (pavolloffay) [inline server/processors-zipkin/src/test/java/org/hawkular/apm/server/processor/zipkin/NodeDetailsDeriverTest.java]: Not necessary timestamp and duration.

File: server/processors-zipkin/src/test/java/org/hawkular/apm/server/processor/zipkin/NodeDetailsDeriverTest.java [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+import org.mockito.Mockito;
+
+/**
+ * @author Gary Brown
+ */
+public class NodeDetailsDeriverTest {
+
+    @Test
+    public void testServerSpanWithURL() throws RetryAttemptException {
+        SpanCache spanCacheMock = Mockito.mock(SpanCache.class);
```
Comment (pavolloffay) [inline server/processors-zipkin/src/test/java/org/hawkular/apm/server/processor/zipkin/NodeDetailsDeriverTest.java]: :+1:

**Comment by objectiser** [comment]
@pavolloffay Timestamp and durations removed. Let me know when ok and I'll rebase before merge.

**Comment by objectiser** [comment]
@pavolloffay Thanks!

**Comment by objectiser** [comment]
retest this please

**Comment by objectiser** [comment]
Merging as build failures seem unrelated. Travis is having trouble finding the jdk, and jenkins is having iptable trouble when trying to run the instrumentation tests. :(

**Pull request merged**